x-pack/libbeat/management/config.go

```diff
@@ -5,112 +5,40 @@
-
-# At debug level, you can selectively enable logging only for some components.
-# To enable all selectors use ["*"]. Examples of other selectors are "beat",
-# "publisher", "service".
-#logging.selectors: ["*"]
-
-#============================== X-Pack Monitoring ===============================
-# {{.BeatName}} can export internal metrics to a central Elasticsearch monitoring
-# cluster.  This requires xpack monitoring to be enabled in Elasticsearch.  The
-# reporting is disabled by default.
-
-# Set to true to enable the monitoring reporter.
-#monitoring.enabled: false
-
-# Uncomment to send the metrics to Elasticsearch. Most settings from the
-# Elasticsearch output are accepted here as well.
-# Note that the settings should point to your Elasticsearch *monitoring* cluster.
-# Any setting that is not set is automatically inherited from the Elasticsearch
-# output configuration, so if you have the Elasticsearch output configured such
-# that it is pointing to your Elasticsearch monitoring cluster, you can simply
-# uncomment the following line.
-#monitoring.elasticsearch:
-`
-
-const (
-	// ModeCentralManagement is a default CM mode, using existing processes.
-	ModeCentralManagement = "x-pack-cm"
-
-	// ModeFleet is a management mode where fleet is used to retrieve configurations.
-	ModeFleet = "x-pack-fleet"
-)
+// ModeFleet is a management mode where fleet is used to retrieve configurations.
+const ModeFleet = "x-pack-fleet"
 
-// Config for central management.
+// Config for central management
 type Config struct {
-	// true when enrolled
-	Enabled bool `config:"enabled" yaml:"enabled"`
-
-	// Mode specifies whether beat uses Central Management or Fleet.
-	// Options: [cm, fleet]
-	Mode string `config:"mode" yaml:"mode"`
-
-	// Poll configs period
-	Period time.Duration `config:"period" yaml:"period"`
-
-	EventsReporter EventReporterConfig `config:"events_reporter" yaml:"events_reporter"`
-
-	AccessToken string `config:"access_token" yaml:"access_token"`
-
-	Kibana *kibana.ClientConfig `config:"kibana" yaml:"kibana"`
-
+	Enabled   bool                    `config:"enabled" yaml:"enabled"`
+	Mode      string                  `config:"mode" yaml:"mode"`
 	Blacklist ConfigBlacklistSettings `config:"blacklist" yaml:"blacklist"`
 }
 
-// EventReporterConfig configuration of the events reporter.
-type EventReporterConfig struct {
-	Period       time.Duration `config:"period" yaml:"period"`
-	MaxBatchSize int           `config:"max_batch_size" yaml:"max_batch_size" validate:"nonzero,positive"`
+// ConfigBlock stores a piece of config from central management
+type ConfigBlock struct {
+	Raw map[string]interface{}
 }
 
+// ConfigBlocksWithType is a list of config blocks with the same type
+type ConfigBlocksWithType struct {
+	Type   string
+	Blocks []*ConfigBlock
+}
+
+// ConfigBlocks holds a list of type + configs objects
+type ConfigBlocks []ConfigBlocksWithType
+
 func defaultConfig() *Config {
 	return &Config{
-		Mode:   ModeCentralManagement,
```

Comment:
I don't think this should have many side effects, but it's worth noting that I changed the default mode (also not sure if we should define a `Mode` attribute and `ModeFleet` constant anymore

Reply:
removed

Reply:
for this removal please update `x-pack/elastic-agent/spec/*.yml` then `mage update` will need to be run.

Reply:
done